File: taller-5-java-ronalcastro97/src/main/java/zooAnimales/Reptil.java
package zooAnimales;
import gestion.*;
import java.util.ArrayList;

public class Reptil extends Animal{
	private static ArrayList<Reptil>listado=new ArrayList<>();
	public static int iguanas=0;
	public static int serpientes=0;
	private String colorEscamas;
	private int largoCola;
	public Reptil() {
		super();
		 Reptil.listado.add(this);
	}
	public Reptil(String nombre, int edad, String habitat, String genero,String colorEscamas,int largoCola) {
		super(nombre, edad, habitat, genero);
		this.colorEscamas=colorEscamas;
		this.largoCola=largoCola;
		Reptil.listado.add(this);
	}
	public static ArrayList<Reptil> getListado() {
		return listado;
	}
	public static void setListado(ArrayList<Reptil> listado) {
		Reptil.listado = listado;
	}
	public static int getIguanas() {
		return iguanas;
	}
	public static void setIguanas(int iguanas) {
		Reptil.iguanas = iguanas;
	}
	public static int getSerpientes() {
		return serpientes;
	}
	public static void setSerpientes(int serpientes) {
		Reptil.serpientes = serpientes;
	}
	public String getColorEscamas() {
		return colorEscamas;
	}
	public void setColorEscamas(String colorEscamas) {
		this.colorEscamas = colorEscamas;
	}
	public int getLargoCola() {
		return largoCola;
	}
	public void setLargoCola(int largoCola) {
		this.largoCola = largoCola;
	}
	public String movimiento(){
		return "reptar";	
	}	
	public static Reptil crearIguana(String nombre,int edad,String genero){
		Reptil iguana= new Reptil(nombre,edad,"humedad",genero,"verde",3);
		Reptil.iguanas++;
		return iguana;
	}
	public static Reptil crearSerpiente(String nombre,int edad,String genero){
		Reptil serpiente= new Reptil(nombre,edad,"jungla",genero,"blanco",1);
		Reptil.serpientes++;
		return serpiente;
	}
	public static int cantidadReptiles() {
		return Reptil.listado.size();
		}
}

File: taller-5-python-ronalcastro97/zooAnimales/reptil.py
from zooAnimales.animal import Animal

class Reptil(Animal):
    _listado=[]
    iguanas=0
    serpientes=0

    def __init__(self, nombre, edad, habitat, genero,colorEscamas,largoCola):
        super().__init__(nombre, edad, habitat, genero)
        self._colorEscamas=colorEscamas
        self._largoCola=largoCola
        Reptil._listado.append(self)
    
    def getColorEscamas(self):
        return self._colorEscamas
    
    def setColorEscamas(self,colorEscamas):
        self._colorEscamas=colorEscamas
    
    def getLargoCola(self):
        return self._largoCola
    
    def setLargoCola(self,largoCola):
        self._largoCola=largoCola

    @classmethod
    def crearIguana(cls,nombre,edad,genero):
        cls.iguanas+=1
        return cls(nombre,edad,"humedal",genero,"verde",3)
    
    @classmethod
    def crearSerpiente(cls,nombre,edad,genero):
        cls.serpientes+=1
        return cls(nombre,edad,"jungla",genero,"blanco",1)
    
    @staticmethod
    def cantidadReptiles():
        return len(Reptil._listado)
     
    @staticmethod
    def movimiento():
        return "reptar"

